packages/shared/src/hooks/useBookmarkPost.ts

```diff
@@ -1,15 +1,19 @@
+import { useContext } from 'react';
 import { useMutation } from 'react-query';
 import request from 'graphql-request';
 import { apiUrl } from '../lib/config';
 import {
   ADD_BOOKMARKS_MUTATION,
   REMOVE_BOOKMARK_MUTATION,
 } from '../graphql/posts';
+import AnalyticsContext from '../contexts/AnalyticsContext';
 
 type MutateFunc<T> = (variables: T) => Promise<(() => void) | undefined>;
 type UseBookmarkPostParams<T> = {
   onBookmarkMutate: MutateFunc<T>;
   onRemoveBookmarkMutate: MutateFunc<T>;
+  onBookmarkTrackObject?: Record<string, unknown>;
+  onRemoveBookmarkTrackObject?: Record<string, unknown>;
```

Comment:
One thing, it's better to have it as a function rather than an actual object.
Something like below. Because otherwise, we may provide some out-of-date values. Providing a function makes sure everything is up to date and generated only when necessary.
```suggestion
  onBookmarkTrackObject?: () => Record<string, unknown>;
  onRemoveBookmarkTrackObject?: () => Record<string, unknown>;
```